lib/3scale/client.rb

```diff
@@ -52,24 +52,37 @@ class Client
     private_constant :EXTENSIONS_HEADER
 
     def initialize(options)
-      if options[:provider_key].nil? || options[:provider_key] =~ /^\s*$/
-        raise ArgumentError, 'missing :provider_key'
+      @provider_key = options[:provider_key]
+      @service_token = options[:service_token]
+      @use_service_tokens = options[:use_service_tokens]
```

Comment:
Well it's already bad that clients have to be stuck with a provider, but it would be worse that if they don't have the provider key they would have to be stuck as well with a specific service. If you wanted to use a different service you would need to create a new client.